### DISCUSSION TITLE
Conditional metada for different profiles on a website

### DISCUSSION THREAD

**Discussion by iuryt** [quarto-dev/quarto-cli#4455] [Q&A]
Hi,

I am trying to write a multilingual web blog using Quarto. I think this software is amazing, but I am having trouble in defining different titles and categories for the blog posts depending on the language. I am using Quarto 1.2.1 CLI.

The post is

```markdown
---
title: Trying Quarto
categories: News
author: "Iury Simoes-Sousa"
date: "2023-02-20"
---


::: {.content-visible when-profile="english"}

This is the first post in a Quarto blog. Welcome!

![](https://quarto.org/quarto.png)

I plan to post here my thoughts about Science, Technology and Music.
:::


::: {.content-visible when-profile="portuguese"}

Este é o meu primeiro post no meu blog usando Quarto. Seja bem vindo!

![](https://quarto.org/quarto.png)

Eu pretendo postar aqui as minhas idéias e tutoriais sobre ciência, tecnologia e música.

:::
```

which returns the following for english

![image](https://user-images.githubusercontent.com/5797727/220523193-06bbe5c3-832c-485e-81c2-e25cbbc4b25f.png)

and the blog page is

```markdown
---
title: "My Blog"
listing:
  contents: posts
  sort: "date desc"
  type: default
  categories: true
  sort-ui: false
  filter-ui: false
page-layout: full
title-block-banner: true
---
```

![image](https://user-images.githubusercontent.com/5797727/220523318-642c1557-942b-4506-8fbe-4bd26fb89d0d.png)

Well, if I use what is suggested in #2822

The post will be:

```markdown
---
title:  |
    ::: {.content-visible when-profile="english"}
    Trying Quarto
    :::
    ::: {.content-visible when-profile="portuguese"}
    Testando o Quarto
    :::
categories: News
author: "Iury Simoes-Sousa"
date: "2023-02-20"
---


::: {.content-visible when-profile="english"}

This is the first post in a Quarto blog. Welcome!

![](https://quarto.org/quarto.png)

I plan to post here my thoughts about Science, Technology and Music.
:::


::: {.content-visible when-profile="portuguese"}

Este é o meu primeiro post no meu blog usando Quarto. Seja bem vindo!

![](https://quarto.org/quarto.png)

Eu pretendo postar aqui as minhas idéias e tutoriais sobre ciência, tecnologia e música.

:::
```

Which returns the following for english:

![image](https://user-images.githubusercontent.com/5797727/220522430-709f80bf-ea2d-4270-bf35-9cc863ec6a74.png)

and it messes up with the Blog page

![image](https://user-images.githubusercontent.com/5797727/220522657-221924de-8cfc-4a17-902b-d202b3fbeac1.png)

**Comment by dragonstyle** [comment]
Unfortunately, we currently assume that the title for a document is a set of inlines (not blocks, as Divs are). As a result, when rendering, we run into errors as you see attempting to render titles as inlines when the block elements are present. This makes some sense when you consider that the title for the page might appear places like meta tags (twitter, open graph) or in RSS feeds where block type elements in titles wouldn't make sense.

I wasn't able to find a great way to conditionalize this while using the banner style title block, unfortunately. This sort of gets you where you're going:

```markdown
---
categories: News
author: "Iury Simoes-Sousa"
date: "2023-02-20"
---

::: {.content-visible when-profile="english"}

# Trying Quarto

This is the first post in a Quarto blog. Welcome!

![](https://quarto.org/quarto.png)

I plan to post here my thoughts about Science, Technology and Music.
:::


::: {.content-visible when-profile="portuguese"}

# Testando o Quarto

Este é o meu primeiro post no meu blog usando Quarto. Seja bem vindo!

![](https://quarto.org/quarto.png)

Eu pretendo postar aqui as minhas idéias e tutoriais sobre ciência, tecnologia e música.

:::
```

But like I said this won't use the banner style title blocks...

**Reply by iuryt** [reply]
Yeah.. I see your point.
What I was looking for was something like

```markdown
---
author: "Iury Simoes-Sousa"
date: "2023-02-20"
---

::: {.content-visible when-profile="english"}
---
categories: News
title: Trying Quarto
---
::: 

::: {.content-visible when-profile="portuguese"}
---
categories: Notícias
title: Testando Quarto
---
::: 
```

But it doesn't work. I think that Quarto should accept something like that, why not?
Maybe this is not exactly to chose when a content will be visible or not, but maybe having a way to set the profile for yaml metadata would be awesome!

**Reply by iuryt** [reply]
Should I transfer this to an issue to request a new feature then?

**Reply by dragonstyle** [reply]
I do think that opening a new issue to track an enhancement makes sense, ty!
